File: food-retail-marketplace/ardo-backend/domain/organization/valueobject/currency.go
package valueobject

import "github/nnniyaz/ardo/pkg/core"

var (
	ErrInvalidCurrency = core.NewI18NError(core.EINVALID, core.TXT_INVALID_CURRENCY)
)

type Currency string

const (
	USD Currency = "USD"
	KZT Currency = "KZT"
	HKD Currency = "HKD"
)

func NewCurrency(currency string) (Currency, error) {
	convertedCurrency := Currency(currency)
	switch convertedCurrency {
	case USD, KZT, HKD:
		return convertedCurrency, nil
	}
	return "", ErrInvalidCurrency
}

func (c Currency) String() string {
	return string(c)
}

File: apps/ardo-marketplace/src/client/domain/base/currency/currency.ts
export enum Currency {
    HKD = "HKD",
    KZT = "KZT",
}

export function ValidateCurrency(currency: string): Error | null {
    if (currency !== Currency.HKD && currency !== Currency.KZT) {
        return new Error("Invalid currency");
    }
    return null;
}
